src/main/java/de/wejago/hichi2influx/service/JsonSubscriber.java

```diff
@@ -0,0 +1,72 @@
+package de.wejago.hichi2influx.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.influxdb.client.domain.WritePrecision;
+import com.influxdb.client.write.Point;
+import de.wejago.hichi2influx.config.Device;
+import de.wejago.hichi2influx.repository.InfluxDbRepository;
+import java.time.Instant;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.map.HashedMap;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class JsonSubscriber extends Subscriber{
+    private final ObjectMapper objectMapper;
+    private final InfluxDbRepository influxDbRepository;
+    private final Device device;
+
+    @Override public void messageArrived(String s, MqttMessage mqttMessage) throws Exception {
+        try {
+            String receivedMessage = new String(mqttMessage.getPayload());
+
+            if(device != null && device.getOnly_match() != null && receivedMessage.contains(device.getOnly_match())) {
+                Map<String, Object> deviceToPointProperties = readDataFromMqttMessage(receivedMessage);
+                Point point = generateMeasurementPoint(deviceToPointProperties);
```

Comment:
I would separate the reading and the mapping, then it is easier to test both.
1. Read -> get string, convert to json, publish to mapper
2. Map -> get Json tree, find values via mapping, create point, write to influx

Reply:
I didn't get what do you propose here. I have private methods that separate the logic as you mention:
```
Map<String, Object> deviceToPointProperties = readDataFromMqttMessage(receivedMessage);
Point point = generateMeasurementPoint(deviceToPointProperties);
influxDbRepository.writePoint(point);
```